verification/src/http_server/handlers/verification/sourcify/api.rs

```diff
@@ -0,0 +1,97 @@
+use crate::{VerificationResponse, VerificationResult};
+use actix_web::{error, error::Error};
+
+use super::types::{ApiFilesResponse, ApiRequest, ApiVerificationResponse, Files};
+
+#[async_trait::async_trait]
+pub(super) trait SourcifyApi {
+    async fn verification(
+        &self,
+        params: &ApiRequest,
+    ) -> Result<ApiVerificationResponse, reqwest::Error>;
+
+    async fn source_files(&self, params: &ApiRequest) -> Result<ApiFilesResponse, reqwest::Error>;
+}
+
+pub(super) struct SoucifyApiClient {
+    host: String,
+}
+
+impl SoucifyApiClient {
+    pub fn new(host: &str) -> Self {
+        Self {
+            host: host.to_string(),
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl SourcifyApi for SoucifyApiClient {
+    async fn verification(
+        &self,
+        params: &ApiRequest,
+    ) -> Result<ApiVerificationResponse, reqwest::Error> {
+        let resp = reqwest::Client::new()
+            .post(&self.host)
+            .json(&params)
+            .send()
+            .await?;
+
+        resp.json().await
+    }
+
+    async fn source_files(&self, params: &ApiRequest) -> Result<ApiFilesResponse, reqwest::Error> {
+        let url = format!(
```

Comment:
if host will be url, then you can do `host.join("files", "any", params.chain, params.address)`

Reply:
or at least this arm

Reply:
I propose this way:
```
let url = self
            .host
            .join(format!("files/any/{}/{}", &params.chain, &params.address).as_str())
            .expect("should be valid url");
```